app/code/Magento/SalesRule/Model/ResourceModel/Rule/DeleteCustomerUsage.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\SalesRule\Model\ResourceModel\Rule;
+
+class DeleteCustomerUsage
+{
+    /** @var \Magento\SalesRule\Model\ResourceModel\Rule\Customer */
+    protected $_customerRuleDetails;
+
+    /**
+     * DeleteCustomerUsage constructor.
+     * @param Customer $customerRuleDetails
+     */
+    public function __construct(
+        Customer $customerRuleDetails
+    ) {
+        $this->_customerRuleDetails = $customerRuleDetails;
+    }
+
+    /**
+     * Delete the time Usage from salesrule_customer table when times_used is 0
+     * @param int $ruleId
+     * @param int $customerId
+     * @param int $updatedTimeUsed
+     */
+    public function execute($ruleId, $customerId, $updatedTimeUsed)
+    {
+        $connection = $this->_customerRuleDetails->getConnection();
+        if ($updatedTimeUsed === 0) {
+            $connection->delete(
+                $connection->getTableName('salesrule_customer'),
```

Comment:
Use `getTable` method of the resource model for correct table name (including prefix if it exists). 
Integration tests failures caused by this issue.

Reply:
@sidolov ,Thank you.
I have made changes. kindly check and let me know.